src/frontend/app/store/effects/api.effects.ts

```diff
@@ -44,95 +44,103 @@ export class APIEffect {
+        // Convert q into a cf q string
+        paginationParams.qString = qParamsToString(paginationParams.q);
+        for (const q of paginationParams.qString) {
+          options.params.append('q', q);
         }
-        for (const key in paginationParams) {
-          if (paginationParams.hasOwnProperty(key)) {
-            if (key === 'page' || !options.params.has(key)) { // Don't override params from actions except page.
-              options.params.set(key, paginationParams[key] as string);
-            }
+        delete paginationParams.qString;
+        delete paginationParams.q;
+      }
+      for (const key in paginationParams) {
+        if (paginationParams.hasOwnProperty(key)) {
+          if (key === 'page' || !options.params.has(key)) { // Don't override params from actions except page.
+            options.params.set(key, paginationParams[key] as string);
           }
         }
-        if (!options.params.has(resultPerPageParam)) {
-          options.params.set(resultPerPageParam, resultPerPageParamDefault.toString());
-        }
       }
+      if (!options.params.has(resultPerPageParam)) {
+        options.params.set(resultPerPageParam, resultPerPageParamDefault.toString());
+      }
+    }
 
-      options.url = `/pp/${proxyAPIVersion}/proxy/${cfAPIVersion}/${options.url}`;
-      options.headers = this.addBaseHeaders(
-        apiAction.cnis ||
-        state.requestData.endpoint, options.headers
-      );
+    options.url = `/pp/${proxyAPIVersion}/proxy/${cfAPIVersion}/${options.url}`;
+    options.headers = this.addBaseHeaders(
+      apiAction.cnis ||
+      state.requestData.endpoint, options.headers
+    );
 
-      if (paginatedAction.flattenPagination) {
-        options.params.set('page', '1');
-      }
+    if (paginatedAction.flattenPagination) {
+      options.params.set('page', '1');
+    }
 
-      let request = this.makeRequest(options);
+    let request = this.makeRequest(options);
 
-      // Should we flatten all pages into the first, thus fetching all entities?
-      if (paginatedAction.flattenPagination) {
-        request = this.flattenPagination(request, options);
-      }
+    // Should we flatten all pages into the first, thus fetching all entities?
+    if (paginatedAction.flattenPagination) {
+      request = this.flattenPagination(request, options);
+    }
 
-      return request
-        .map(resData => this.handleMultiEndpoints(resData, apiAction)) // Check for errors and fetch entities
-        .mergeMap(response => {
-          const { entities, totalResults, totalPages } = response;
-          const actions = [];
-          actions.push({ type: apiAction.actions[1], apiAction });
-          actions.push(new WrapperRequestActionSuccess(
-            entities,
-            apiAction,
-            requestType,
-            totalResults,
-            totalPages
-          ));
+    return request
+      .withLatestFrom(Observable.of({ ...paginatedAction }).first())
```

Comment:
Is this still needed? Doesn't look like _action is used later on. Otherwise it should be actionClone